  ---
title: "Confirmed ABS patient taxonomy analysis"
output: html_notebook
---

```{r setup, include=FALSE}
library(openxlsx)
library(readxl)
library(ggpubr)
library(ggplot2)
library(dplyr)
library(reshape2)
library(stringr)
library(vegan)
library(devtools)
library(forcats)
library(ggbiplot)
library(plotly)
library(rstatix)
setwd("C:/Users/Cynthia/OneDrive - University of California, San Diego Health/#Schnabl/#AutoBrewery Syndrome/Metagenomics/Taxonomy analysis")
meta <- read.csv('C:/Users/Cynthia/OneDrive - University of California, San Diego Health/#Schnabl/#AutoBrewery Syndrome/Metagenomics/ABS_confirmedonly_metadata.csv')
```

# PHYLA TAXONOMY ANALYSIS
## Split up Phyla data
```{r}
df <- read.csv("Batch1to6andHC_metaphlan3_abundance_table.csv")
no_class <- dplyr::filter(df, !grepl("c__", Species))
phyla <- dplyr::filter(no_class, grepl("p__", Species))
bacteria_phyla <- dplyr::filter(phyla, grepl("k__Bacteria", Species))
#bacteria_phyla$Species <- gsub("k__Bacteria|p__","", bacteria_phyla$Species)
#bacteria_phyla$Species <- gsub("[^[:alnum:][:blank:]+?&/\\-]","", bacteria_phyla$Species)
```

## Making relative abundance for phyla from raw counts
```{r}
phylum <- bacteria_phyla[,1]
phyla1 <- apply(as.matrix(bacteria_phyla[,-1]), 2, as.numeric)
df.rel <- prop.table(phyla1, margin = 2)

df.rel <- as.data.frame(df.rel)
df.rel <- cbind(phylum, df.rel)

df.rel2 = setNames(data.frame(t(df.rel[,-1])), df.rel[,1])
ID <- rownames(df.rel2)
df.rel3 <- cbind(ID, data.frame(df.rel2, row.names=NULL))

phyla2 <- merge(meta[,1:11], df.rel3,by=c("ID"))
phyla2$Other <- phyla2$Bacteriaunclassified + phyla2$CandidatusMelainabacteria +
  phyla2$Fusobacteria + phyla2$Synergistetes + phyla2$Tenericutes
phyla3 = subset(phyla2, select = -c(Bacteriaunclassified, CandidatusMelainabacteria, 
                                    CandidatusSaccharibacteria, Spirochaetes, Lentisphaerae,
                                    Fusobacteria, Synergistetes, Tenericutes))
```


## FIGURE 2A Waterfall plot showing up to 100%
```{r}
df.long <- phyla3 %>% 
  gather(variable, value, 12:17)

df.long2 <- df.long[,c(1,6,7,11,12,13)]

avg_phyla <- df.long2 %>%
  group_by(Flare, House, BAC, variable) %>%
  summarise(
    mean = mean(value, na.rm = TRUE)
  )

df.short <- avg_phyla %>%
  spread(variable, mean)
df.short$diff <- df.short$Firmicutes-df.short$Proteobacteria
df.short1 <- df.short[order(df.short$diff, decreasing = TRUE),]

df.long3 <- df.short1 %>% 
  gather(variable, value, 4:9)


df.long3$Flare <- factor(df.long3$Flare, c("HC", "HHP","Remission", "Flare"))
df.long3$variable <- factor(df.long3$variable, c("Firmicutes", "Actinobacteria", "Bacteroidetes", "Other",  "Verrucomicrobia", "Proteobacteria"))
df.long3$ID2 <- paste(df.long3$House, df.long3$Flare)
df.long3 <- subset(df.long3, House != "A080")

ggplot(df.long3, aes(x = fct_inorder(ID2), y = value, fill = variable)) +
  geom_col(position = "fill")+
  facet_grid(~Flare, scales = "free_x", space = "free_x")+
  theme_classic()+
  rremove("x.ticks")+
  scale_y_continuous(name = "Relative Abundance", labels = scales::percent)+
  scale_fill_brewer(palette = "Spectral")+
  scale_x_discrete(labels = function(x) str_wrap(x, width = 8))+
  theme(text = element_text(colour = "black", size=11),
        axis.text=element_text(colour="black", size=11),
        axis.title.x = element_blank(),
        legend.title = element_blank(),
        axis.text.x = element_text(angle = 90, vjust = 0.5))

ggsave(
  "Taxa_phyla_barplot_7.20.2024.png",
  width = 1800,
  height = 720,
  units = "px",
  dpi = 200
) 
# 1000 x 360
```


## FIGURE 2B Phyla relative abundance HHP vs ABS
```{r}
mycomp <- list(c("Remission", "Flare"), c("HHP", "Flare"), c("HC", "Flare"))
df.long4 <- subset(df.long3, variable != "Other")

ggboxplot(
  df.long4, x="Flare", y="value", 
  add = c("jitter"), 
  facet.by = "variable", nrow = 1,
  fill = "Flare", palette = c("#40557B", "#007561", "#ff6361", "#ffa600"))+
  stat_compare_means(aes(group = Flare),label.y = 1.25) +
  stat_compare_means(comparisons = mycomp)+
  scale_y_continuous(name = "Relative Abundance", labels = scales::percent)+
  rremove("x.ticks")+
  theme(axis.title.x = element_blank(),
        legend.position = "none",
        text = element_text(colour = "black", size=12),
        axis.text=element_text(colour="black", size=12))
        
ggsave(
  "Taxa_phyla_boxplots_7.20.2024.png",
  width = 2000,
  height = 720,
  units = "px",
  dpi = 200
) 


ggpaired(df.long4, id = "House", x="Flare", y="value",
         line.color = "gray", line.size = 0.4, 
         facet.by = "variable", nrow = 1,
         fill = "Flare", palette = c("#40557B", "#007561", "#ff6361", "#ffa600"))+
  stat_compare_means(aes(group = Flare),label.y = 1.25) +
  stat_compare_means(comparisons = mycomp)+
  scale_y_continuous(name = "Relative Abundance", labels = scales::percent)+
  rremove("x.ticks")+
  theme(axis.title.x = element_blank(),
        legend.position = "none",
        text = element_text(colour = "black", size=12),
        axis.text=element_text(colour="black", size=12))

# 1200 x 360
```


## Proteobacteria relative abundance
```{r}
avg_phyla <- df.long2 %>%
  group_by(Flare, House, variable) %>%
  summarise(
    mean = mean(value, na.rm = TRUE)
  )

df.short <- avg_phyla %>%
  spread(variable, mean)

df.short$Flare <- factor(df.short$Flare, c("HC", "HHP","Remission", "Flare"))
mycomp <- list(c("Remission", "Flare"), c("HHP", "Flare"))
ggboxplot(df.short, id = "House", x="Flare", y="Proteobacteria",
          add = "jitter",
         fill = "Flare", palette = c("#40557B", "#007561", "#ff6361", "#ffa600"))+
  stat_compare_means(aes(group = Flare),label.y = 1.0) +
  stat_compare_means(comparisons = mycomp)+
  scale_y_continuous(name = "Relative Abundance: Proteobacteria", labels = scales::percent)+
  rremove("x.ticks")+
  theme(axis.title.x = element_blank(),
        legend.position = "none",
        text = element_text(colour = "black", size=12),
        axis.text=element_text(colour="black", size=12))

ggsave(
  "Proteobacteria_bxplot_7.20.2024.png",
  width = 720,
  height = 720,
  units = "px",
  dpi = 200
) 
```


# FIGURE 2C Correlation with BAC
```{r}
ggscatter(df.short, x = "BAC", y = "Proteobacteria",
          add = "reg.line",                                 # Add regression line
          conf.int = TRUE,                                  # Add confidence interval
          add.params = list(color = "black",
                            fill = "lightgray"),
          )+
  xlab("Blood alcohol concentration (mg/dL)")+
  scale_y_continuous(name = "Relative abundance: Proteobacteria", labels = scales::percent)+
  stat_cor(method = "spearman", label.x = 3, label.y = 1)  # Add correlation coefficient

ggsave(
  "BAC_proteobacteria_corr_7.20.2024.png",
  width = 900,
  height = 720,
  units = "px",
  dpi = 200
) 
# 480 x 400
```
