---
title: "Maaslin2 for Humann3 results"
author: "Cynthia Hsu"
date: "2024-07-03"
output: html_document
---

```{r setup, include=FALSE}
#Load MaAsLin 2 package into the R environment
library(Maaslin2)
library(dplyr)
library(ggpubr)
library(ggplot2)
library(reshape2)
library(tidyr)
library(stringr)
library(openxlsx)
setwd("/Volumes/Seagate Portable Drive/#MetagenomicSequencing/##AUD/output_merged")
```

# merge metagenomic data files together
```{r}
all.paths <- as.data.frame(read.delim('all_pathabundance_relab.tsv', sep="\t"))

AUD <- read.csv('AUD_pathabundance_relab.csv')
HC <- read.csv('HC_pathabundance_relab.csv')
HC_AUD <- merge(HC, AUD, by = 'ID', all.x = T, all.y = T)
HC_AUD[is.na(HC_AUD)] <- 0
#write.xlsx(HC_AUD, 'HC_AUD_pathabundance_relab.xlsx')

AUD <- read.csv('AUD_metacyc_genefamilies_relab.csv')
HC <- read.csv('HC_metacyc_genefamilies_relab.csv')
HC_AUD <- merge(HC, AUD, by = 'ID', all.x = T, all.y = T)
HC_AUD[is.na(HC_AUD)] <- 0
write.xlsx(HC_AUD, 'HC_AUD_metacyc_genefamilies_relab.xlsx')
```

## Cleaning data to include only the pathway
```{r}
df <- read_excel('HC_AUD_pathabundance_relab.xlsx')
df1 <- dplyr::filter(df, !grepl("UNMAPPED",ID))
df2 <- dplyr::filter(df1, !grepl("UNINTEGRATED",ID))
df3 <- dplyr::filter(df2, !grepl("g__",ID))
df4 <- dplyr::filter(df3, !grepl("unclassified",ID))
df5 = setNames(data.frame(t(df4[,-1])), df4$ID)
ID <- rownames(df5)
df6 <- cbind(ID, data.frame(df5, row.names=NULL))
path_abundance <- df6
rownames(path_abundance) <- ID
path_abundance <- path_abundance[,-1]
```

## Running MaAsLin2 with min abundance and prevalence
```{r}   
meta1 <- read.csv('PS_Metadata.csv')
input_metadata <- subset(meta1, Dx == "AUD" | Dx == "HC")
rownames(input_metadata) <- input_metadata$ID

fit_dataFlare  = Maaslin2(input_data = path_abundance, 
                     input_metadata = input_metadata,
                     min_abundance  = 0.00001,
                     min_prevalence = 0.2,
                     normalization  = "NONE",
                     output         = "pathabundance20_Dx2_againstAUD1_7.16.2024",
                     fixed_effects  = "Dx2",
                     random_effects = "Patient",
                     reference      = 'Dx2,AUD_W1',
                     max_significance = 0.10,
                     standardize = FALSE)
```

# NOW METACYC PATHWAYS
## Cleaning data to include only the pathway
```{r}
df <- read_excel('HC_AUD_metacyc_genefamilies_relab.xlsx')
df1 <- dplyr::filter(df, !grepl("UNMAPPED",ID))
df2 <- dplyr::filter(df1, !grepl("UNINTEGRATED",ID))
df3 <- dplyr::filter(df2, !grepl("g__",ID))
df4 <- dplyr::filter(df3, !grepl("unclassified",ID))
df5 = setNames(data.frame(t(df4[,-1])), df4$ID)
ID <- rownames(df5)
df6 <- cbind(ID, data.frame(df5, row.names=NULL))
path_abundance <- df6
rownames(path_abundance) <- ID
path_abundance <- path_abundance[,-1]
```

## Running MaAsLin2 with min abundance and prevalence
```{r}   
meta1 <- read.csv('PS_Metadata.csv')
input_metadata <- subset(meta1, Dx == "AUD" | Dx == "HC")
rownames(input_metadata) <- input_metadata$ID

fit_dataFlare  = Maaslin2(input_data = path_abundance, 
                     input_metadata = input_metadata,
                     min_abundance  = 0.00001,
                     min_prevalence = 0.2,
                     normalization  = "NONE",
                     output         = "pathabundance20_Dx2_metacyc_genefam_7.16.2024",
                     fixed_effects  = c("Dx2"),
                     random_effects = "Patient",
                     reference      = c('Dx2,HC'),
                     max_significance = 0.10,
                     standardize = FALSE)
```







# CODE FROM ABS PROJECT, NOT OPTIMIZED FOR AUD YET
## Combine with pathway lineage classification, filter only for significant results, re-order
```{r}
pathways <- read.csv('all_results_house_HC_7.14.csv')
pathways2 <- pathways %>%
  separate_wider_delim(feature, "..", names = c("Pathway", "Name"), too_many = "merge")
class <- read.csv('map_metacyc-pwy_lineage.csv')
class1 <- class[!duplicated(class[1]),]
path1 <- merge(class1, pathways2, by = 'Pathway', all.y = T)

allpaths <- as.list(path1$Pathway)
allpaths <- unique(allpaths)


important_paths <- data.frame()


### USE THIS CRITERIA
for (path in allpaths) {
  if (
      ((subset(path1, (value == "Remission") & Pathway == path, select = qval) < 0.025) &
       (abs(subset(path1, value == "Remission" & Pathway == path, select = coef)) > 1)) |
      ((subset(path1, (value == "HC") & Pathway == path, select = qval) < 0.01) &
       (abs(subset(path1, value == "HC" & Pathway == path, select = coef)) > 2.5)) |
      ((subset(path1, (value == "HHP") & Pathway == path, select = qval) < 0.01) &
       (abs(subset(path1, value == "HHP" & Pathway == path, select = coef)) > 2.1))) {
    important_paths <- rbind(important_paths, path1[path1$Pathway == path,])
  }
}
###


### USE THIS CRITERIA2
for (path in allpaths) {
  if (
      (
        (subset(path1, (value == "Remission") & Pathway == path, select = qval) < 0.1) &
        (subset(path1, value == "Remission" & Pathway == path, select = coef)) > 1)|
      (
        (((subset(path1, (value == "HC") & Pathway == path, select = qval) < 0.1) &
        (subset(path1, value == "HC" & Pathway == path, select = coef)) > 1)) &
      (subset(path1, (value == "HHP") & Pathway == path, select = qval) < 0.1)&
        (subset(path1, value == "HC" & Pathway == path, select = coef)) > 1)) {
    important_paths <- rbind(important_paths, path1[path1$Pathway == path,])
  }
}
###

#### DIDN'T USE THIS CRITERIA
for (path in allpaths) {
  if ((subset(path1, value == "HHP" & Pathway == path, select = qval) < 0.08) &
      (abs(subset(path1, value == "Remission" & Pathway == path, select = coef)) > 0.8)) {
    important_paths <- rbind(important_paths, path1[path1$Pathway == path,])
  }
  else if((subset(path1, value == "HHP" & Pathway == path, select = qval) < 0.01) &
      (abs(subset(path1, value == "HHP" & Pathway == path, select = coef)) > 2.5) &
      (abs(subset(path1, value == "Remission" & Pathway == path, select = qval)) < 0.8)) {
    important_paths <- rbind(important_paths, path1[path1$Pathway == path,])
  }
}
####

important_paths <- subset(important_paths, !is.na(Lineage))

important_paths2 <- important_paths %>%
  separate_wider_delim(Lineage, "|", names = c("Main", "Sub"), too_many = "drop", too_few = "align_start")

important_paths3 <- important_paths2[order(important_paths2$Main, important_paths2$Sub),]
important_paths3$Name <- gsub('\\.', ' ', important_paths3$Name)

write.csv(important_paths3, "significant.pathways.classified_7.14.csv")

```

## Plotting Maaslin2 results as a bar plot

```{r}
  
ggplot(important_paths3, aes(x=fct_rev(fct_inorder(Name)), y= as.numeric(-coef),
                   fill=as.numeric(qval)))+ 
  geom_bar(stat="identity", position=position_dodge())+
  scale_fill_gradient2(low="#132B4B", mid = "#E2F1FF", high="white", midpoint = 0.5)+
  facet_grid(~value)+
  coord_flip() +
  theme_classic()+
  theme(text = element_text(colour = "black", size=11),
        axis.text=element_text(colour="black", size=11),
        axis.title.x = element_blank(),
        axis.title.y = element_blank())

ggsave(
  "Maaslin2_barplot_7.14.2024.png",
  width = 2000,
  height = 1500,
  units = "px",
  dpi = 200
) 

```

## Boxplots of highlighted pathways

```{r}
# merge meta and df6
meta <- read.csv('C:/Users/Cynthia/OneDrive - University of California, San Diego Health/#Schnabl/#AutoBrewery Syndrome/Metagenomics/ABS_confirmedonly_metadata.csv')
meta <- subset(meta, !is.na(House))
df60 <- merge(meta[,c(1,4,6,11)], df6, by = "newID", )

# average by household
avg.df60 <- df60 %>% 
  group_by(House, Flare) %>%
  summarise_if(is.numeric, mean)


avg.df60$Flare <- factor(avg.df60$Flare, levels = c("HC", "HHP", "Remission", "Flare"))


# Mixed acid fermentation
mycomp <- list(c("Remission", "Flare"), c("HHP", "Flare"))
ggpaired(avg.df60, id = "House", x="Flare", y="FERMENTATION.PWY..mixed.acid.fermentation",
         line.color = "gray", line.size = 0.4, 
         fill = "Flare", palette = c("#40557B", "#007561", "#ff6361", "#ffa600"))+
  stat_compare_means(label.y = 4.5E-04, paired = T)+
  stat_compare_means(comparisons = mycomp)+
  ylab("Mixed acid fermentation pathway")+ 
  rremove("x.ticks")+
  rremove("x.title")+
  theme(text = element_text(colour = "black", size=12),
        axis.text=element_text(colour="black", size=12),
        legend.position = "none")

ggsave(
  "Mixed acid fermentation_boxplot_7.14.2024.png",
  width = 800,
  height = 760,
  units = "px",
  dpi = 200
) 

### 360x360

# Heterolactic fermentation
mycomp <- list(c("Remission", "Flare"), c("HHP", "Flare"), c("HC", "Flare"))
ggpaired(avg.df60, id = "House", x="Flare", y="P122.PWY..heterolactic.fermentation",
         line.color = "gray", line.size = 0.4, 
         fill = "Flare", palette = c("#40557B","#007561", "#ff6361", "#ffa600"))+
  stat_compare_means(label.y = 1.16E-04, paired = T)+
  stat_compare_means(comparisons = mycomp)+
  ylab("Heterolactic fermentation pathway")+ 
  rremove("x.ticks")+
  rremove("x.title")+
  theme(text = element_text(colour = "black", size=12),
        axis.text=element_text(colour="black", size=12),
        legend.position = "none")

ggsave(
  "Heterolactic fermentation_boxplot_7.14.2024.png",
  width = 800,
  height = 760,
  units = "px",
  dpi = 200
) 

# Ethanolamine utilization 
mycomp <- list(c("Remission", "Flare"), c("HHP", "Flare"), c("HC", "Flare"))
ggpaired(avg.df60, id = "House", x="Flare", y="PWY0.1477..ethanolamine.utilization",
         line.color = "gray", line.size = 0.4, 
         fill = "Flare", palette = c("#40557B", "#007561", "#ff6361", "#ffa600"))+
  stat_compare_means(label.y = 10E-04, paired = T)+
  stat_compare_means(comparisons = mycomp)+
  ylab("Ethanolamine utilization pathway")+ 
  rremove("x.ticks")+
  rremove("x.title")+
  theme(text = element_text(colour = "black", size=12),
        axis.text=element_text(colour="black", size=12),
        legend.position = "none")

ggsave(
  "Ethanolamine utilization_boxplot_7.14.2024.png",
  width = 800,
  height = 760,
  units = "px",
  dpi = 200
) 

# TCA acetate producers

ggpaired(avg.df60, id = "House", x="Flare", y="PWY.7254..TCA.cycle.VII..acetate.producers.",
         line.color = "gray", line.size = 0.4, 
         fill = "Flare", palette = c("#40557B", "#007561", "#ff6361", "#ffa600"))+
  stat_compare_means(label.y = 4.8E-05, paired = T)+
  stat_compare_means(comparisons = mycomp)+
  ylab("TCA cycle (acetate producers)")+ 
  rremove("x.ticks")+
  rremove("x.title")+
  theme(text = element_text(colour = "black", size=12),
        axis.text=element_text(colour="black", size=12),
        legend.position = "none")
```


## Plot different fermentation pathways stacked 
```{r}
df5 <- df1 %>% filter(
  newID %in% c("FERMENTATION-PWY: mixed acid fermentation"
               ,"PWY0-1477: ethanolamine utilization"
               ,"P122-PWY: heterolactic fermentation"
               ))

df6 = setNames(data.frame(t(df5[,-1])), df5[,1])
newID <- rownames(df6)
df7 <- cbind(newID, data.frame(df6, row.names=NULL))

df8<- merge(meta, df7, by = "newID")

means <- df8 %>% 
  group_by(House, Flare) %>%
  summarise_if(is.numeric, mean)

means <- subset(means, !is.na(House))

long_DF <- means %>% 
  gather(Pathway, rel_abundance, 5:6
         )
         
long_DF$Flare <- factor(long_DF$Flare, levels = c("HHP", "Remission", "Flare"))
long_DF$Pt <- paste(long_DF$House, long_DF$Flare)

ggplot(long_DF, aes(x= reorder(Pt, rel_abundance),y = rel_abundance, fill=Pathway),
       palette = "spectral") + 
  geom_bar(position="stack", stat="identity") +
  facet_grid(~Flare, scales = "free_x", space = "free_x")+
  scale_fill_brewer(palette = "Spectral")+
  ylab("Fermentation pathway enrichment")+
  rremove("x.title")+
  geom_hline(yintercept=6.0675e-05, linetype="dashed", color = "black")+
  theme_classic()+
  theme(text = element_text(colour = "black", size=11),
        axis.text=element_text(colour="black", size=11),
        axis.text.x = element_text(angle = 90, vjust = 0.5),
        axis.ticks.x = element_blank())

```



# PFAM - extra code

## looking at Fe-ADH PFAM 	PF00465

```{r}
df <- read.csv('all_genefamilies_uniref90_pfam.csv')
df.adh <- df[grep("PF00465",df$newID),]
df.adh$sum <- rowSums(df.adh>0)
df.adh1 <- subset(df.adh, sum>5)
df.adh2 <- subset(df.adh1, select = -c(sum) )
df.adhT = setNames(data.frame(t(df.adh2[,-1])), df.adh2[,1])
newID <- rownames(df.adhT)
df.adh <- cbind(newID, data.frame(df.adhT, row.names=NULL))

df.adhmerge <- merge(input_metadata, df.adh)
mycomp <- list(c("Remission", "Flare"), c("HHP", "Flare"))

long_DF <- df.adhmerge %>% gather(Feature, rel_abundance, PF00465:PF00465.unclassified)
long_DF$Flare <- factor(long_DF$Flare, levels = c("HHP", "Remission", "Flare"))
ggboxplot(long_DF, x="Flare", y="rel_abundance", fill = "Flare",
          add = "jitter", palette = c("#e6f598", "#66c2a5", "#5e4fa2"),
          facet.by = "Feature", nrow = 7) +
  stat_compare_means(label.y = 0.002)+
  stat_compare_means(comparisons = mycomp)+
  ylab("Relative abundance")+ 
  rremove("x.ticks")+
  rremove("x.title")+
  theme(text = element_text(colour = "black", size=12),
        axis.text=element_text(colour="black", size=12),
        legend.position = "none")

```